lib/core/profile.jl

```
@@ -0,0 +1,261 @@
+        # Without this, two separate capture managers are potentially handled
+        # One with capture enabled and one without
+        device()
+        handle = mtSharedCaptureManager()
+        obj = new(handle)
+        # No finalizer needed since the manager is handled by Metal
+        return obj
+    end
+end
+
+Base.unsafe_convert(::Type{MTLCaptureManager}, capman::MtlCaptureManager) = capman.handle
+
+Base.:(==)(a::MtlCaptureManager, b::MtlCaptureManager) = a.handle == b.handle
+Base.hash(capman::MtlCaptureManager, h::UInt) = hash(capman.handle, h)
+
+"""
+    startCapture(obj::Union{MtlDevice,MtlCommandQueue}, destination::MtCaptureDestination=MtCaptureDestinationGPUTraceDocument; folder::String=nothing)
+
+Start GPU frame capture using the default capture object and specifying capture descriptor parameters directly.
+"""
+function startCapture(obj::Union{MtlDevice,MtlCommandQueue}, destination::MtCaptureDestination=MtCaptureDestinationGPUTraceDocument; folder::String=nothing)
+    destination == MtCaptureDestinationGPUTraceDocument && folder == nothing && throw(ArgumentError("Must specify output folder if destination is GPUTraceDocument"))
+    startCapture(MtlCaptureManager(), MtlCaptureDescriptor(obj, destination; folder=folder))
+end
+
+"""
+    startCapture(manager::MtlCaptureManager, desc::MtlCaptureDescriptor)
+
+Start a GPU frame capture session with the given capture manager and descriptor.
+"""
+function startCapture(manager::MtlCaptureManager, desc::MtlCaptureDescriptor)
+    _errptr = Ref{MTLError}()
+    result = mtStartCaptureWithDescriptor(manager.handle, desc.handle, _errptr)
+    if !(result)
+        err = MtlError(_errptr[])
+        # Provide more informative error messages for unsupported capture
+        err.domain == "MTLCaptureError" && err.code == 1 && println("Error: Did you set the Metal capture environment variable before launching Julia? \
```

Use `"""` instead of escaping endlines.

---

Also, why the `println` and `throw`? We support nested exceptions nowadays, so you can just throw an exception capturing the child MtlError.